File: services/reservation/src/main/java/com/example/iter/reservation/dto/response/EvidenceImagePresignResponse.java
package com.example.iter.reservation.dto.response;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

public record EvidenceImagePresignResponse(
        List<Item> uploads
) {
    public record Item(
            String objectKey,
            String uploadUrl,
            Map<String, String> requiredHeaders,
            String publicUrl,
            LocalDateTime expiresAt
    ) {
    }
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/dto/response/EvidenceImagePresignResponse.kt
package com.example.iter.reservation.dto.response

import java.time.LocalDateTime

data class EvidenceImagePresignResponse(
    val uploads: List<Item>,
) {
    fun uploads(): List<Item> = uploads

    data class Item(
        val objectKey: String,
        val uploadUrl: String,
        val requiredHeaders: Map<String, String>,
        val publicUrl: String?,
        val expiresAt: LocalDateTime,
    ) {
        fun objectKey(): String = objectKey
        fun uploadUrl(): String = uploadUrl
        fun requiredHeaders(): Map<String, String> = requiredHeaders
        fun publicUrl(): String? = publicUrl
        fun expiresAt(): LocalDateTime = expiresAt
    }
}
